examples/virtual-dom-wip/platform/Json.roc

```diff
@@ -1205,79 +1205,79 @@ expect
-                                # TODO This doesn't seem right, shouldn't we eat
-                                # the remaining json object value bytes if we are skipping this
-                                # field?
-                                { result: Ok recordState, rest: valueBytes }
-
-                            Keep valueDecoder ->
-                                # Decode the value using the decoder from the recordState
-                                # Note we need to pass json config options recursively here
-                                Decode.decodeWith valueBytes valueDecoder (@Json {})
-                    )
-                    |> tryDecode \{ val: updatedRecord, rest: bytesAfterValue } ->
-                        # Check if another field or '}' for end of object
-                        when List.walkUntil bytesAfterValue (AfterObjectValue 0) objectHelp is
-                            ObjectFieldNameStart n ->
-                                rest = List.dropFirst bytesAfterValue n
-
-                                # Decode the next field and value
-                                decodeFields updatedRecord rest
-
-                            AfterClosingBrace n ->
-                                rest = List.dropFirst bytesAfterValue n
-
-                                # Build final record from decoded fields and values
-                                when finalizer updatedRecord json is
-                                    Ok val -> { result: Ok val, rest }
-                                    Err e -> { result: Err e, rest }
-
-                            _ ->
-                                # Invalid object
-                                { result: Err TooShort, rest: bytesAfterValue }
-
-        countBytesBeforeFirstField =
-            when List.walkUntil bytes (BeforeOpeningBrace 0) objectHelp is
-                ObjectFieldNameStart n -> n
-                _ -> 0
+        when objectNameResult is
+            Err TooShort ->
+                # Invalid object, unable to decode field name or find colon ':'
+                # after field and before the value
+                { result: Err TooShort, rest: bytes }
 
-        if countBytesBeforeFirstField == 0 then
-            # Invalid object, expected opening brace '{' followed by a field
-            { result: Err TooShort, rest: bytes }
-        else
-            bytesBeforeFirstField = List.dropFirst bytes countBytesBeforeFirstField
+            Ok objectName ->
+                # Decode the json value
+                (
+                    fieldName = objectName
+
+                    # Retrieve value decoder for the current field
+                    when stepField recordState fieldName is
+                        Skip ->
+                            # TODO This doesn't seem right, shouldn't we eat
+                            # the remaining json object value bytes if we are skipping this
+                            # field?
```

Comment:
TODO

Reply:
This was an existing TODO; all of this code just got unindented 4 spaces.